### DISCUSSION TITLE
postgREST log add customs fields

### DISCUSSION THREAD

**Discussion by surendirababu-r** [PostgREST/postgrest#3152] [Q&A] [answered]
**Hi all , I need one help
This is my postgREST log format**

127.0.0.1 - postgres [10/Jan/2024:10:55:27 +0530] "GET /subscription_user_role_mapping?UserId=eq.110000 HTTP/1.1" 200 - "http://localhost:3000/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
127.0.0.1 - postgres [10/Jan/2024:10:55:28 +0530] "GET /subscription_client_users?ClientId=eq.1100001 HTTP/1.1" 200 - "http://localhost:3000/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"

**I need to add userid in this log like**
127.0.0.1 - postgres [10/Jan/2024:10:55:28 +0530] "GET /subscription_client_users?ClientId=eq.1100001 HTTP/1.1" 200 - "http://localhost:3000/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36" "1100001"

**that user id i send headers userid filed

is there any options available for add new filed in log format ?**

**Comment by steve-chavez** [comment]
> 127.0.0.1 - postgres [10/Jan/2024:10:55:28 +0530] "GET /subscription_client_users?ClientId=eq.1100001 HTTP/1.1" 200 - "http://localhost:3000/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36" "1100001"

Hm, so you want to duplicate the query parameter (`1100001`) already shown on the URL? Why? Can you elaborate your use case?

**Reply by surendirababu-r** [reply]
127.0.0.1 - postgres [10/Jan/2024:10:55:28 +0530] 
request url - "GET /subscription_client_users?ClientId=eq.1100001 HTTP/1.1" 
status - 200  
dev url -"http://localhost:3000/" 
user agent - "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36" 
user id - "23hu39pl"


query parameter id & user id not same for every request also i don't send (ClientId=eq.1100001) for every request i just send login user id in headers that id need to add in postgREST log

**Reply by steve-chavez** [reply, marked as answer]
Ah, so you want to surface some headers to the log line.

For now you could do it with Nginx like so https://ruslan.rocks/posts/nginx-log-headers

For PostgREST itself, there's currently now way to do this. Maybe open an issue requesting a config similar to Nginx's `log_format`.

**Reply by surendirababu-r** [reply]
ok now understand thanks' brother
